### PR TITLE
Create missing systemd accounts

### PR DESCRIPTION
**Pull request by tgopinath-microsoft**
Presently, systemd-oom and systemd-coredump service accounts are not created during RPM install and are created on initial boot. Create these users and groups during installation like other generic systemd accounts.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x ] The toolchain/worker package manifests are up-to-date
- [x ] Any updated packages successfully build (or no packages were changed)
- [x ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x ] All package sources are available
- [x ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x ] All source files have up-to-date hashes in the `*.signatures.json` files
- [x ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x ] Documentation has been updated to match any changes to the build system
- [x ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
systemd-oom and systemd-coredump are core systemd accounts that are not being generated during install. Due to this we ran into a bug in the following scenario
-  systemd tries to create these accounts during boot
- In between the system gets power down
- On the subsequent boot /etc/passwd is corrupted.

This is also a systemd bug where writes to /etc/passwd were not being fsynced properly and that is fixed as part of a separate  PR. Nevertheless these missing accounts should be created during install

###### Change Log  

N/A

###### Does this affect the toolchain?  

YES

###### Associated issues  

- https://github.com/Azure/azure-sphere-v2-are-fs/issues/57

###### Links to CVEs  
N/A

###### Test Methodology

- Sphere Pipeline : https://dev.azure.com/msazuresphere/Public/_build/results?buildId=291802&view=results

### CODE REVIEW

File: SPECS/filesystem/filesystem.spec [hunk 1]
```diff
@@ -704,6 +708,9 @@ return 0
 %config(noreplace) /etc/modprobe.d/tipc.conf
 
 %changelog
+* Thu Sep 14 2022 Thara Gopinath <tgopinath@microsoft.com> - 1.1-12
+- Create missing systemd user and group accounts
```
Comment (PawelWMS) [inline SPECS/filesystem/filesystem.spec:712]: Small request: let's explicitly list the added accounts, so it's easier to see what changed just by reading the changelog.

```suggestion
- Add the 'systemd-coredump' and 'systemd-oom' user accounts.
```

**Review by PawelWMS — APPROVED** [review]
One request mentioned in the comments, otherwise looks good. Thanks for the change, @tgopinath-microsoft!

**Comment by tgopinath-microsoft** [comment]
Thanks for the review ! Have incorporated your suggestion

**Comment by tgopinath-microsoft** [comment]
@PawelWMS : The spec linting check is throwing up a lot of things wrong with filesystem.spec that I have not touched.. Is this something I should be fixing with this PR?

**Comment by dmcilvaney** [comment]
> @PawelWMS : The spec linting check is throwing up a lot of things wrong with filesystem.spec that I have not touched.. Is this something I should be fixing with this PR?

Not Pavel, but: we do try to keep the linter happy, but partly advisory, especially for some of the more complicated specs. I'd leave your changes minimal. We can maybe clean up the spec next time we do a big change to it.

**Comment by tgopinath-microsoft** [comment]
Thanks Daniel.. I will leave it as is then

**Comment by oliviacrain** [comment]
Cleared the merge conflict, will merge momentarily.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**